File: BackEnd/Bateaux/src/main/java/org/nst/bateaux/dto/reservation/ReservationAdd.java
package org.nst.bateaux.dto.reservation;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.nst.bateaux.entity.StatusRes;
import org.nst.bateaux.entity.TypeReservation;

import java.time.LocalDateTime;
@Data
@AllArgsConstructor
@NoArgsConstructor

public class ReservationAdd {
    LocalDateTime date ;
    TypeReservation typeReservation;
    int nbPersonnes;
    StatusRes status;
}

File: Back-office/boat-rental-admin/src/app/interfaces/reservation-add.ts
export interface ReservationAdd {
  date: string;
  typeReservation: 'FULL_DAY' | 'HALF_DAY' | 'TWO_HOURS';
  nbPersonnes: number;
  status?: 'EN_ATTENTE' | 'ACCEPTER' | 'REFUSER';
}
